ovos_core/skill_installer.py

```diff
@@ -0,0 +1,174 @@
+            return False
+        # Use constraints to limit the installed versions
+        if constraints and not exists(constraints):
+            LOG.error('Couldn\'t find the constraints file')
+            self.play_error_sound()
+            return False
+        elif exists(SkillsStore.DEFAULT_CONSTRAINTS):
+            constraints = SkillsStore.DEFAULT_CONSTRAINTS
+
+        pip_args = [sys.executable, '-m', 'pip', 'install']
+        if constraints:
+            pip_args += ['-c', constraints]
+        if self.config.get("break_system_packages", False):
+            pip_args += ["--break-system-packages"]
+
+        with SkillsStore.PIP_LOCK:
+            """
+            Iterate over the individual Python packages and
+            install them one by one to enforce the order specified
+            in the manifest.
+            """
+            for dependent_python_package in packages:
+                LOG.info("(pip) Installing " + dependent_python_package)
+                pip_command = pip_args + [dependent_python_package]
+                LOG.debug(" ".join(pip_command))
+                if print_logs:
+                    proc = Popen(pip_command)
+                else:
+                    proc = Popen(pip_command, stdout=PIPE, stderr=PIPE)
+                pip_code = proc.wait()
+                if pip_code != 0:
+                    stderr = proc.stderr.read().decode()
+                    self.play_error_sound()
+                    raise RuntimeError(stderr)
+
+        reload(ovos_plugin_manager)  # force core to pick new entry points
+        self.play_success_sound()
+        return True
+
+    def pip_uninstall(self, packages: list,
+                      print_logs: bool = True):
+        if not len(packages):
+            self.play_error_sound()
+            return False
+
+        pip_args = [sys.executable, '-m', 'pip', 'uninstall']
+        if self.config.get("break_system_packages", False):
+            pip_args += ["--break-system-packages"]
+
+        with SkillsStore.PIP_LOCK:
+            """
+            Iterate over the individual Python packages and
+            install them one by one to enforce the order specified
+            in the manifest.
+            """
+            for dependent_python_package in packages:
+                LOG.info("(pip) Uninstalling " + dependent_python_package)
+                pip_command = pip_args + [dependent_python_package]
+                LOG.debug(" ".join(pip_command))
+                if print_logs:
+                    proc = Popen(pip_command)
+                else:
+                    proc = Popen(pip_command, stdout=PIPE, stderr=PIPE)
+                pip_code = proc.wait()
+                if pip_code != 0:
+                    stderr = proc.stderr.read().decode()
+                    self.play_error_sound()
+                    raise RuntimeError(stderr)
+
+        reload(ovos_plugin_manager)  # force core to pick new entry points
```

Comment:
Is it only possible to do this with everything at once?

Reply:
seems to get all installed skills to properly reload, but if you find a better way let me know!